Add zhihe-series 4G modem stick support

185264646 opened the pull request:
May close the old pull request for ufi001b made by pho. I've edited the dts as the comments given in that pull request.
The reason why I renamed the dts is decribed in the pull request in [https://github.com/msm8916-mainline/lk2nd/pull/223](url).
Notice: there are some other boards similar to this one but don't begin with ufi, so use ufi as the vendor name is not very appropriate. I prefer use entire `ufi-001c` as the name of the board.

185264646 committed:
dt-bindings: vendor-prefixes: add zhihe

zhihe is a placeholder for various 4G USB dongles made by unknown
manufactures with similar design. It is widely used in China to refer to
these dongles.

Signed-off-by: Yang Xiwen <forbidden405@foxmail.com>

stephan-gh commented:
If you have used @pho's original PR as a base then please preserve the original author of their commit. You could use something like:

```
From: Jaime Breva <jbreva@nayarsystems.com>

Signed-off-by: Jaime Breva <jbreva@nayarsystems.com>
Co-developed-by: Your Name <your@name.com>
Signed-off-by: Your Name <your@name.com>
```

If you have made significant changes to everything and the majority of changes is actually from you, you can swap the main author (make yourself the author and pho/Jaime as Co-developed-by).

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts:
```diff
@@ -0,0 +1,260 @@
+			no-map;
+		};
+
+		gps_mem: gps@8bc00000 {
+			reg = <0x0 0x8bc00000 0x0 0x200000>;
```
0x86800000 + 0x5500000 = 0x8bd00000

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts:
```diff
@@ -0,0 +1,260 @@
+
+	reserved-memory {
+		mpss_mem: mpss@86800000 {
+			reg = <0x00 0x86800000 0x00 0x5500000>;
```
```suggestion
			reg = <0x0 0x86800000 0x0 0x5500000>;
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts:
```diff
@@ -0,0 +1,260 @@
+&usb_hs_phy {
+	extcon = <&pm8916_usbin>;
+};
+
+&usb {
+	status = "okay";
+	extcon = <&pm8916_usbin>;
+	dr_mode = "peripheral";
+};
+
+&pm8916_usbin {
+	status = "okay";
+};
+
+&sdhc_1 {
+	status = "okay";
+
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&sdc1_clk_on &sdc1_cmd_on &sdc1_data_on>;
+	pinctrl-1 = <&sdc1_clk_off &sdc1_cmd_off &sdc1_data_off>;
+};
+
+&pronto {
+	status = "okay";
+};
+
+&bam_dmux {
+	status = "okay";
+};
+
+&bam_dmux_dma {
+	status = "okay";
+};
+
+&mpss {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&sim_ctrl>;
+};
```
These nodes would look better if sorted alphabetically.

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts:
```diff
@@ -0,0 +1,260 @@
+};
+
+&msmgpio {
+	usb_id: usb-id {
+		pins = "gpio110";
+		function = "gpio";
+
+		drive-strength = <8>;
+		bias-pull-up;
+	};
```
This pinctrl doesn't seem to be used.

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts:
```diff
@@ -0,0 +1,260 @@
+		bias-pull-up;
+	};
+
+	button: button {
+		pins = "gpio37";
+		function = "gpio";
+
+		bias-pull-down;
+		drive-strength = <2>;
+	};
+
+	gpio_leds: gpio-leds {
+		pins = "gpio20", "gpio21", "gpio22";
+		function = "gpio";
+
+		bias-disable;
+		drive-strength = <2>;
+	};
+
+	sim_ctrl: sim-ctrl {
+		pins = "gpio1", "gpio2";
+		function = "gpio";
+
+		bias-disable;
+		output-low;
+		drive-strength = <2>;
+	};
```
`*_default: *-default-state` is preferred. See 8b276ca036377a5baa4fd0692b80608f0de8a260

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts:
```diff
@@ -0,0 +1,260 @@
+
+		label = "GPIO Buttons";
+
+		restart {
```
```suggestion
		button-restart {
```

b08f5cbd69dcd25f5ab2a0798fe3836a97a9d7c6

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts:
```diff
@@ -0,0 +1,42 @@
+
+/ {
+	model = "ufi-001c/ufi-001b 4G Modem Stick";
+	compatible = "zhihe, ufi001c", "qcom,msm8916";
```
```suggestion
	compatible = "zhihe,ufi001c", "qcom,msm8916";
```

`<vendor>,<codename>`  (without spaces)

wonderfulShrineMaidenOfParadise reviewed (commented):
Looks much better now. Yet you have to squash the small commits into large ones, and have 2 large commits in the end:
1. `arm64: dts: qcom: msm8916-ufi-1c: Add initial dts`
2. `arm64: dts: qcom: msm8916-ufi-1c: Enable modem`

Remember to sign-off with your formal name and email, so we can finally upstream the commits to the mailing lists.

wonderfulShrineMaidenOfParadise commented:
Consider joining:
https://matrix.to/#/#msm8939:matrix.org for MSM8916 specific supports
https://matrix.to/#/#mainline:postmarketos.org for generic mainline kernel supports

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts:
```diff
@@ -0,0 +1,42 @@
+	};
+
+	leds {
+		led_r {
```
```suggestion
		led-r {
```

Looks like a typo.

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts:
```diff
@@ -0,0 +1,42 @@
+			gpios = <&msmgpio 22 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_g {
```
```suggestion
		led-g {
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts:
```diff
@@ -0,0 +1,42 @@
+			gpios = <&msmgpio 21 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_b {
```
```suggestion
		led-b {
```

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts:
```diff
@@ -0,0 +1,42 @@
+	model = "ufi-001c/ufi-001b 4G Modem Stick";
+	compatible = "zhihe,ufi001c", "qcom,msm8916";
+	
+	gpio_keys {
```
```suggestion
	gpio-keys {
```

Dashes for node names and underscores for labels.

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts:
```diff
@@ -0,0 +1,42 @@
+
+/ {
+	model = "ufi-001c/ufi-001b 4G Modem Stick";
+	compatible = "zhihe,ufi001c", "qcom,msm8916";
```
```suggestion
	compatible = "zhihe,ufi001c", "qcom,msm8916";
	chassis-type = "embedded";
```

Even if it's not the best pick, probably the closest from what the spec defines.
I suppose the value could be omitted but spec says "Optional but Recommended" so I think it's better to have it.

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts:
```diff
@@ -0,0 +1,42 @@
+	model = "ufi-001c/ufi-001b 4G Modem Stick";
+	compatible = "zhihe,ufi001c", "qcom,msm8916";
+	
+	gpio-keys {
```
Prefer using labels to extend existing nodes, this way it's obvious some proeprties are inherited from another file

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-zhihe-ufi001c.dts:
```diff
@@ -0,0 +1,42 @@
+	};
+
+	leds {
+		led-r {
```
same about having a label for each led. Also if there are some boards that have no leds/ different set of leds maybe the whole node could be defined in the "variant-specific" dts

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi:
```diff
@@ -0,0 +1,237 @@
+		pinctrl-names = "default";
+		pinctrl-0 = <&gpio_leds_default>;
+
+		led-r {
```
These leds have "cellular" and "wifi" icons near them on the housing, so maybe use i.e. wan and wlan functions for them, so it's obvious which is which when configuring the userspace?

185264646 replied:
> These leds have "cellular" and "wifi" icons near them on the housing, so maybe use i.e. wan and wlan functions for them, so it's obvious which is which when configuring the userspace?

I don't see any icons at least on my own device's housing except a wifi icon, but all leds are under that icon so it's quite meaningless. Generally, every device uses red led to indicate kernel is up, but I'm not sure about green led and blue led.

TravMurav replied:
Oh, interesting, I thought they all look somewhat like this with two led "areas":
![изображение](https://user-images.githubusercontent.com/22081178/210396144-70d381d6-6692-4d3b-96c7-ee6ba403d856.png)

You can ignore the comment then and keep it as-is

185264646 replied:
> Oh, interesting, I thought they all look somewhat like this with two led "areas": ![изображение](https://user-images.githubusercontent.com/22081178/210396144-70d381d6-6692-4d3b-96c7-ee6ba403d856.png)
> 
> You can ignore the comment then and keep it as-is

Maybe I can put some comments in dtsi to suggest others that what function should they give to each led, but I'm not sure.

TravMurav replied:
nah, don't think it's necessary to annotate in this case, we can just override these properties when we add the next board with the leds marked

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi:
```diff
@@ -0,0 +1,237 @@
+};
+
+&sdhc_1 {
+	status = "okay";
```
Nowdays upstream wants to have status= always being the last proeprty, so let's do that from the start. (2 places)

(I was working on cleaning those all up across this repo and it's pretty fun to fixup over 100 commits with each of them causing at least one merge conflict later...)

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi:
```diff
@@ -0,0 +1,237 @@
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&button_default>;
```
```suggestion
		pinctrl-0 = <&button_default>;
		pinctrl-names = "default";
```
I recently learned that upstream prefers to have -names after the "numbered" properties so let's follow that. 4 places to change.

185264646 replied:
> I recently learned that upstream prefers to have -names after the "numbered" properties so let's follow that. 4 places to change.

Should have no missed nodes.

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi:
```diff
@@ -0,0 +1,238 @@
+		};
+	};
+
+	gpio_keys: gpio-keys {
```
Is this label used?

185264646 replied:
> Is this label used?

No, `leds` is also unused right now. Should they be removed?

TravMurav replied:
Unless you can justify a reason to have it, since it will be "compiled out" at build time with no uses anyway

TravMurav reviewed (commented):
This looks good to me so let's squash the commits now.

I propose making two commits: Adding the board, and enabling the modem. This separation is because the modem bits are not yet in the mainline linux, but everything else that the board uses is.

This way the second commit would add the modem.dtsi as well as the relevant reserved-memory, mpss and the sim pinctrls.

I suggest you to annotate both commit the way @stephan-gh has proposed (You can decide if you consider your changes significant enough to put yourself as a primary author or as a co-author for the commits, I think both are fine here)

Please remember that upstream wants all commits to be authored by people using their real name, and they forbid using aliases. Saying that, I'm unable to verify the name you have used.

I'd really appreciate to see a commit message on the first commit, you can look at these for example:
https://github.com/msm8916-mainline/linux/commit/55056b229189be2b4b8e636f0566a0b5bfd3c8f8
https://github.com/msm8916-mainline/linux/commit/16e8e8072108426029f0c16dff7fbe77fae3df8f

185264646 committed:
arm64: dts: qcom: msm8916-zhihe: Add initial device tree for zhihe Wifi/LTE dongle UFI-001C and uf896

This commit adds support for the ufi-001C and uf896 WiFi/LTE dongle made by
unknown manufactures based on MSM8916.
uf896 is another variant for the zhihe usb stick. The board design
difers by using different gpios for the keys and leds.

Note: The original firmware does not support 64-bit OS. It is necessary
to flash 64-bit TZ firmware to boot arm64.

Currently supported:
- All CPU cores
- Buttons
- LEDs
- Modem
- SDHC
- USB Device Mode
- UART

Co-developed-by: Jaime Breva <jbreva@nayarsystems.com>
Signed-off-by: Jaime Breva <jbreva@nayarsystems.com>
Co-developed-by: Nikita Travkin <nikita@trvn.ru>
Signed-off-by: Nikita Travkin <nikita@trvn.ru>
Signed-off-by: Yang Xiwen <forbidden405@foxmail.com>

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi:
```diff
@@ -0,0 +1,238 @@
+			no-map;
+		};
+
+		gps_mem: gps@8bd00000 {
```
The user for gps_mem is also in the msm8916-modem.dtsi (since gps is handled by modem on 8916)
https://github.com/msm8916-mainline/linux/blob/msm8916/6.2-rc2/arch/arm64/boot/dts/qcom/msm8916-modem.dtsi#L26-L28

Let's have the whole reserved-memory node added in the second commit

TravMurav reviewed (commented):
On the commit messages:

#### Add initial device tree:
Thanks for writing a description!
The tags are a bit dubious:
- I don't see Jaime's `Co-developed-by` tag. It should go right before their S-o-b.
- The review suggestions aren't as significant as i.e. your work of rewriting someone's patchset. Drop 3 other co-developed-by tags.
- `Reviewed-by` tags ([as most other tags in fact](https://www.kernel.org/doc/html/v4.17/process/submitting-patches.html#when-to-use-acked-by-cc-and-co-developed-by)) are usually *given* by a person that did the review. You should not add them unless asked to.
  This tag usually means the reviewer has "approved" the patch and has no further comments. For this repo we use review ui and PR approves instead, so, if anywhere, you will gain the review tags on the Linux kernel lists upstream. Drop them.
- It's expected that the author has tested their patchset. Please omit the Tested-by tag under your name.
- From checkpatch.pl:
```
WARNING: From:/Signed-off-by: email name mismatch: 'From: Xiwen Yang <forbidden405@foxmail.com>' != 'Signed-off-by: Yang Xiwen <forbidden405@foxmail.com>'
```
Usually you would use `git commit -s` (or `--signoff') to add a tag with the same name as the one in your git config. You can set the name per-repo as well.

#### Enable modem:

- checkpatch:
```
ERROR: Missing Signed-off-by: line by nominal patch author 'Jaime Breva <jbreva@nayarsystems.com>'
```

Usually we would have the CI to run the checkpatch, but sadly it seems to be dead lately and we haven't replaced it just yet.

TravMurav commented:
fwiw this is a patch with uf896:
https://gist.githubusercontent.com/TravMurav/3205fa4e3d0985f058ee994568aaf088/raw/c40d597f09b7789578617ad44ff963bfc65e512a/0001-arm64-dts-qcom-msm8916-zhihe-uf896-Add-uf896-usb-don.patch

You can add it to your tree by running `curl <link> | git am`. Feel free to either add it as-is as (before the modem commit) or, better, squash into the first and add me as a co-author (remember to keep the s-o-b of each co-autor right after their tag and sort them in the order of "time", so if you do, mine would be the second)

I decided to not push it to your tree since you re-use a branch from msm8916-mainline so that might get a bit annoying.

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi:
```diff
@@ -0,0 +1,238 @@
+/dts-v1/;
+
+#include "msm8916-pm8916.dtsi"
+#include "msm8916-modem.dtsi"
```
After discussing with @stephan-gh I propose dropping the modem dtsi and enabling the modem/network right here instead since we have no benefit from the audio setup or gps anyway. This way, 100% of the file will be immediately upstreamable.
Just add
```
&bam_dmux {
	status = "okay";
};

&bam_dmux_dma {
	status = "okay";
};

&mpss {
	status = "okay";
};
```
where appropriate and squash all commits into a single one.

185264646 replied:
> After discussing with @stephan-gh I propose dropping the modem dtsi and enabling the modem/network right here instead since we have no benefit from the audio setup or gps anyway. This way, 100% of the file will be immediately upstreamable. Just add
> 
> ```
> &bam_dmux {
> 	status = "okay";
> };
> 
> &bam_dmux_dma {
> 	status = "okay";
> };
> 
> &mpss {
> 	status = "okay";
> };
> ```
> 
> where appropriate and squash all commits into a single one.

I know somebody is using gps to track his model car, Maybe it is worthwhile to setup gps.

TravMurav replied:
GPS depends on the "memshare" driver that is a bit "questionable" at least in terms of upstreaming, so I'd like to separate it for now. (we can still have it in msm8916-mainline of course)

You can add memshare + memory region as a separate commit straight into your dts (instead of using the modem.dtsi)

TravMurav reviewed (commented):
This looks pretty good for me, since the goal of dropping the dtsi was to squash the commits into one, let's do that. Please also fix the tags like:
```
Co-developed-by: Jaime Breva <jbreva@nayarsystems.com>
Signed-off-by: Jaime Breva <jbreva@nayarsystems.com>
Co-developed-by: Nikita Travkin <nikita@trvn.ru> # uf896
Signed-off-by: Nikita Travkin <nikita@trvn.ru>
Signed-off-by: Yang Xiwen <forbidden405@foxmail.com>
```
(this would clearly show why people sign-off the commit, otherwise it would mean "Jamie gave the commit (that was made by Yang) to Nikita without changes, Nikita passed it to Yang, also without changes")

Then you can add the memshare/gps as a separate commit, adding ~~the `gps_mem` as well as~~ these in it:
```
&memshare {
	status = "okay";
};

&memshare_gps {
	memory-region = <&gps_mem>;
};
```

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi:
```diff
@@ -0,0 +1,247 @@
+			reg = <0x0 0x86800000 0x0 0x5500000>;
+			no-map;
+		};
+
+		gps_mem: gps@8bd00000 {
+			reg = <0x0 0x8bd00000 0x0 0x200000>;
+			no-map;
+		};
```
```suggestion
```

Usually a reserved memory added is either to block some memory from being accessed (i.e. if some higher level firmware there is protected, so touching it would result in a crash) or when there is something that needs some specific memory to work (i.e. to load some firmware or data and/or pass it to some other core after that)

gps_mem is closer to the second case, the special memshare driver gives this memory to modem, that then uses it for gps. This means that without the memshare driver using the memory, this node just takes away 2mb for nothing. Modem is fine without having it, just the gps wouldn't work (some modem firmware can even use gps without memshare)

Since the memshare driver is not upstream yet, let's drop this node in the first commit, and then, if you wish, introduce it in a dedicated "enable memshare" commit (as suggested earlier) that we would keep in msm8916-mainline until we figure out how to properly add memshare stuff upstream.

stephan-gh replied:
I think I was going to add the gps_mem upstream already so we don't need to patch up all devices again later. Would also allow possibly implementing memshare in userspace.

TravMurav replied:
Ah, cool, then I suppose we can keep it as-is

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi:
```diff
@@ -0,0 +1,247 @@
+	l11 {
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <2950000>;
+		regulator-allow-set-load;
+		regulator-system-load = <200000>;
```
```suggestion
		regulator-system-load = <200000>;
		regulator-allow-set-load;
```

Apparently upstream has a preference even in the order of these two...

TravMurav commented on arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi:
```diff
@@ -0,0 +1,247 @@
+&msmgpio {
+	button_default: button-default-state {
+		function = "gpio";
+
```
```suggestion
```
... And they also don't like the separation line in the pinctrl nodes nowdays...

Will have to fix this in all other dts'es... \**sigh\**

TravMurav reviewed (commented):
Adding a couple nitpicks I just got from a review of some my patches upstream, with that fixed and the unused leds/buttons labels removed, this dts should be good

wonderfulShrineMaidenOfParadise commented on arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi:
```diff
@@ -0,0 +1,247 @@
+
+		pinctrl-0 = <&button_default>;
+		pinctrl-names = "default";
+
+		label = "GPIO Buttons";
```
```suggestion
```

Preferred by upstream maintainers.

stephan-gh replied:
I think Konrad clarified that it's not a problem (`and he seems to have confused it with input-name`)?

TravMurav replied:
This is fine, just misunderstanding

TravMurav commented:
In addition to comments above, I'd appreciate if you can add the compatible and the vendor prefix to the documentation. See these commits for the reference:
https://github.com/msm8916-mainline/linux/commit/0b894c4667e084b25a740327e3aac6afe5ae08b2
https://github.com/msm8916-mainline/linux/commit/dc1b8ce9db8c5181b3345505a66dfe8136b83f21

185264646 committed:
dt-bindings: qcom: Ducument bindings for new msm8916-zhihe-uf896 and msm8916-zhihe-ufi001c

Document the new zhihe,uf896/ufi001c device tree bindings used in
their device trees.

Signed-off-by: Yang Xiwen <forbidden405@foxmail.com>

wonderfulShrineMaidenOfParadise commented on Documentation/devicetree/bindings/vendor-prefixes.yaml:
```diff
@@ -1458,6 +1458,8 @@ patternProperties:
     description: Xiaomi Technology Co., Ltd.
   "^xillybus,.*":
     description: Xillybus Ltd.
+  "^signalinks,.*":
```
Not sure about Signalinks but we need to sort by codename.
It should be possible to ask for models from customer support.

185264646 replied:
> Not sure about Signalinks but we need to sort by codename. It should be possible to ask for models from customer support.

sorry for the missing, I renamed it from xinfeng but forgot to sort

stephan-gh commented on Documentation/devicetree/bindings/vendor-prefixes.yaml:
```diff
@@ -1147,6 +1147,8 @@ patternProperties:
   "^sigma,.*":
     description: Sigma Designs, Inc.
+  "^signalinks,.*":
+    description: Signalinks Communication Technology Co., Ltd
```
As mentioned I'd like to avoid "blaming" signalinks for making these sticks if we're not sure they're really the manufacturer.  As far as I'm concerned using `zhihe,` would be fine if that's your preferred name. Please describe what it means in the `description` e.g.:

```yaml
  "^zhihe,.*":
    description: Placeholder for various 4G USB dongles made by unknown manufacturers
```

Ideally find a better word than "Placeholder", because I suspect someone might complain about that word.

stephan-gh commented on arch/arm64/boot/dts/qcom/msm8916-signalinks-stick.dtsi:
```diff
@@ -0,0 +1,247 @@
+		label = "GPIO Buttons";
+
+		button_restart: button-restart {
+			label = "Restart";
```
I'd appreciate a comment here that explains why `gpio` is missing. Even something very simple would be enough:

```
/* gpio is board-specific */
```

which tells the reader to search for the property somewhere else (in the board .dts)

stephan-gh commented on arch/arm64/boot/dts/qcom/msm8916-signalinks-stick.dtsi:
```diff
@@ -0,0 +1,247 @@
+
+		led_r: led-r {
+			color = <LED_COLOR_ID_RED>;
+			default-state = "on";
```
same here for the gpios

stephan-gh commented on arch/arm64/boot/dts/qcom/msm8916-signalinks-stick.dtsi:
```diff
@@ -0,0 +1,247 @@
+	status = "okay";
+};
+
+/* Remove &dsi_phy0 from clocks to make sure that gcc probes */
```
> that gcc probes **with display disabled**

might make the problem more obvious

stephan-gh commented on arch/arm64/boot/dts/qcom/msm8916-signalinks-uf896.dts:
```diff
@@ -0,0 +1,43 @@
+
+&button_default {
+	pins = "gpio35";
+	/delete-property/ bias-pull-down;
```
You have two boards, one of them needs `bias-pull-down`, the other `bias-pull-up`. Clearly the `bias-pull-down` should be in `ufi001c.dts` and not the common .dtsi :)  
Then you can remove the `/delete-property/` here.

stephan-gh commented on arch/arm64/boot/dts/qcom/msm8916-signalinks-uf896.dts:
```diff
@@ -0,0 +1,43 @@
+};
+
+&sim_ctrl_default {
+	pins = "gpio1", "gpio2";
```
Do you know some board where the SIM control GPIOs are different? Right now they are the same for the two dts you're adding so you could add it in the .dtsi.

185264646 replied:
> Do you know some board where the SIM control GPIOs are different? Right now they are the same for the two dts you're adding so you could add it in the .dtsi.

Yes, I know at least 2 boards using other gpios for sim control. Though I'm not going to add support for them now because I don't have device and can not ensure that they work.

stephan-gh replied:
OK, keep it as-is then :)

TravMurav reviewed (commented):
I unresolved some nits that were set resolved (probably by mistake). With those fixed, I think this series would be very nice to send to the lists.

If you wonder about the dt guidelines, Konrad (Who is a reviewer for qcom dts upstream) has shared his dt review notes that you can find here:
https://github.com/konradybcio-work/dt_review/

stephan-gh commented on arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi:
```diff
@@ -0,0 +1,248 @@
+		pinctrl-0 = <&gpio_leds_default>;
+		pinctrl-names = "default";
+
+		/* gpio is board-specific */
```
If you do it on the top-level it's probably clearer to make this

```suggestion
		/* GPIOs are board-specific */
```

(I meant adding it to every single LED, but it's probably fine here on the top-level as well.

stephan-gh commented on arch/arm64/boot/dts/qcom/msm8916-zhihe.dtsi:
```diff
@@ -0,0 +1,248 @@
+};
+
+&msmgpio {
+	/* pins is board-specific */
```
```suggestion
	/* pins are board-specific */
```

stephan-gh reviewed (commented):
Thanks for taking time to address all the review comments. :)
Just a small nitpick remaining, I guess @TravMurav will talk to you about potential upstreaming of the patches (i.e. sending them to the mailing lists).

pho commented:
Thanks @185264646 and everyone involved, for picking up the PR and turn it into something actually upstreamable :)

TravMurav commented:
Imported 3/3 of the series sent upstream into the 6.2-rc3 branch. Once again, big thanks to you and everyone involved!
